### DISCUSSION TITLE
What is the name of the installed binary?

### DISCUSSION THREAD

**Discussion by anasouardini** [ImageMagick/ImageMagick#6325] [Help]
I think the package is installed just fine, but I can't getting to run by typing `imagemagick` (which is the best name in my openion).

If there was no bug while installation, the binary's name is not the same as the package's name in the debian repo, in my openion it only makes the process of using it harder. Any shortening, if that's the purpose of changing the name, can always be done by aliasing.

OS: debian testing.

**Comment by torontal7** [comment]
In Knoppix 9.1, which is based on Debian GNU/Linux bullseye:

$ dpkg -s imagemagick
Package: imagemagick
Status: install ok installed
Priority: optional
Section: oldlibs
Installed-Size: 218
Maintainer: ImageMagick Packaging Team <pkg-gmagick-im-team@lists.alioth.debian.org>
Architecture: i386
Multi-Arch: foreign
Source: imagemagick (8:6.9.11.24+dfsg-1)
Version: 8:6.9.11.24+dfsg-1+b2
Depends: imagemagick-6.q16 (>= 8:6.9.2.10+dfsg-2~)
Description: image manipulation programs -- binaries
 ImageMagick is a software suite to create, edit, and compose bitmap images.
 It can read, convert and write images in a variety of formats (over 100)
 including DPX, EXR, GIF, JPEG, JPEG-2000, PDF, PhotoCD, PNG, Postscript,
 SVG, and TIFF. Use ImageMagick to translate, flip, mirror, rotate, scale,
 shear and transform images, adjust image colors, apply various special
 effects, or draw text, lines, polygons, ellipses and Bézier curves.
 All manipulations can be achieved through shell commands as well as through
 an X11 graphical interface (display).
 .
 This package include links to channel depth specific binaries and manual
 pages.
 .
 This is a dummy package.  You can safely purge or remove it.
Homepage: https://www.imagemagick.org/

[List](https://commandmasters.com/commands/dpkg-linux/) the files installed by imagemagick package:

$ dpkg -L imagemagick
/.
/usr
/usr/share
/usr/share/bug
/usr/share/doc
/usr/share/doc/imagemagick
/usr/share/doc/imagemagick/NEWS.Debian.gz
/usr/share/doc/imagemagick/changelog.Debian.gz
/usr/share/doc/imagemagick/changelog.Debian.i386.gz
/usr/share/doc/imagemagick/changelog.gz
/usr/share/doc/imagemagick/copyright
/usr/share/bug/imagemagick

Working commands:

$ which display convert
/usr/bin/display
/usr/bin/convert

Does not work:

$ which magick imagemagick
$

Print the canonicalized version of convert (it is a symbolic link ):

$ readlink -f /usr/bin/convert
/usr/bin/convert-im6.q16

Find the owner package:

$ dpkg -S /usr/bin/convert-im6.q16
imagemagick-6.q16: /usr/bin/convert-im6.q16

$ dpkg -S /usr/bin/display-im6.q16
imagemagick-6.q16: /usr/bin/display-im6.q16

So, if you see a tutorial using "magick" command, like "[ImageMagick Examples -- Creating Thumbnails and Framing](https://usage.imagemagick.org/thumbnails/)", replace "magick" with "convert". There are also tutorials using "convert" command: "[Resizing Images and Creating Thumbnails with ImageMagick](https://gigi.nullneuron.net/gigilabs/resizing-images-and-creating-thumbnails-with-imagemagick/)". Life would be very boring without such puzzles...

$ dpkg -L imagemagick-6.q16
/.
/usr
/usr/bin
/usr/bin/animate-im6.q16
/usr/bin/compare-im6.q16
/usr/bin/composite-im6.q16
/usr/bin/conjure-im6.q16
/usr/bin/convert-im6.q16
/usr/bin/display-im6.q16
/usr/bin/identify-im6.q16
/usr/bin/import-im6.q16
/usr/bin/mogrify-im6.q16
/usr/bin/montage-im6.q16
/usr/bin/stream-im6.q16
/usr/lib
...

$ dpkg -s imagemagick-6.q16
...
